### PR TITLE
backupmeta: add cluster_version and br_version info in backupmeta (#734)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #734 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In br repo:
git pr https://github.com/pingcap/br/pull/803
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/br.git pr/803:release-4.0-a0018eaa041a
```

---

### What problem does this PR solve? 
https://github.com/pingcap/br/issues/698

### What is changed and how it works?
add `cluster_version` and `br_version` info in backupmeta when we backup data.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has persistent data change

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - add cluster_version and br_version info in backupmeta

### CODE REVIEW

File: go.mod1 [hunk 1]
```diff
@@ -23,7 +23,11 @@ require (
 	github.com/pingcap/check v0.0.0-20200212061837-5e12011dc712
 	github.com/pingcap/errors v0.11.5-0.20201126102027-b0a155152ca3
 	github.com/pingcap/failpoint v0.0.0-20200702092429-9f69995143ce
+<<<<<<< HEAD
 	github.com/pingcap/kvproto v0.0.0-20201126113434-70db5fb4b0dc
+=======
+	github.com/pingcap/kvproto v0.0.0-20210204074845-dd36cf2e1c6b
```
Comment (overvenus) [inline go.mod1:29]: Please use a release-4.0 kvproto.

Comment (kennytm) [reply go.mod1:29]: @lichunzhu ↑

Comment (lichunzhu) [reply go.mod1:29]: @overvenus @kennytm 
Could you please help merge this https://github.com/pingcap/kvproto/pull/727?

**Comment by lichunzhu** [comment]
@kennytm @overvenus PTAL

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**